src/requests/cache/basic/set.rs

```diff
@@ -0,0 +1,68 @@
+use crate::cache_client::CacheClient;
+use crate::requests::cache::MomentoRequest;
+use crate::simple_cache_client::prep_request_with_timeout;
+use crate::{IntoBytes, MomentoResult};
+use std::time::Duration;
+
+/// ```
+/// # fn main() -> anyhow::Result<()> {
+/// # use momento_test_util::create_doctest_client;
+/// # tokio_test::block_on(async {
+/// # let (cache_client, cache_name) = create_doctest_client();
+/// use momento::requests::cache::basic::set::Set;
+///
+/// let set_response = cache_client.set(&cache_name, "key", "value").await?;
+/// assert_eq!(set_response, Set {});
+/// # Ok(())
+/// # })
+/// #
+/// }
+/// ```
+pub struct SetRequest<K: IntoBytes, V: IntoBytes> {
+    cache_name: String,
+    key: K,
+    value: V,
+    ttl: Option<Duration>,
+}
+
+impl<K: IntoBytes, V: IntoBytes> SetRequest<K, V> {
+    pub fn new(cache_name: String, key: K, value: V) -> Self {
+        let ttl = None;
+        Self {
+            cache_name,
+            key,
+            value,
+            ttl,
+        }
+    }
+
+    pub fn with_ttl(mut self, ttl: Duration) -> Self {
+        self.ttl = Some(ttl);
+        self
+    }
+}
+
+impl<K: IntoBytes, V: IntoBytes> MomentoRequest for SetRequest<K, V> {
+    type Response = Set;
+
+    async fn send(self, cache_client: &CacheClient) -> MomentoResult<Set> {
+        // let ttl = self.ttl.unwrap_or_default();
```

Comment:
Is this leftover from set add elements?

Reply:
oops probably, thanks

Reply:
fixed in https://github.com/momentohq/client-sdk-rust/pull/185/commits/a82609ceedbd3336bc109c3f2b3454a1e1d9dcad